scenes/autoload/data_manager.gd

```diff
@@ -16,79 +16,79 @@ var pending_file_changes :Dictionary = {}
 ## save changes to disk at the end of current update frame
 var auto_apply_changes := true
 
-class SectionWrapper:
-	var file_path:String
-	var section:String
-	func set_data(key: String, value: Variant):
-		DataManager.set_data(file_path, section, key, value)
-	func get_data(key: String, default: Variant = null) -> Variant:
-		return DataManager.get_data(file_path, section, key, default)
-	func get_keys() -> PackedStringArray:
-		return DataManager.get_config(file_path).get_section_keys(section)
-
 func _ready() -> void:
 	for file_path in LOAD_ON_READY_PATHS:
-		get_config(file_path)
+		_get_config(file_path)
 
 func get_wrapper(file_path: String, section: String) -> SectionWrapper:
 	var wrapper := SectionWrapper.new()
 	wrapper.file_path = file_path
 	wrapper.section = section
 	return wrapper
 
-func get_config(file_path: String) -> ConfigFile:
-	if not file_data.has(file_path):
-		file_data[file_path] = load_file(file_path)
-	return file_data[file_path]
+func get_data(file_path: String, section: String, key :String, default :Variant = null) -> Variant:
+	var config := _get_config(file_path)
+	return config.get_value(section, key, default)
 
 func set_data(file_path: String, section: String, key: String, value: Variant):
-	var config := get_config(file_path)
+	var config := _get_config(file_path)
 	config.set_value(section, key, value)
 	pending_file_changes[file_path] = true
-	if auto_apply_changes and not get_tree().process_frame.is_connected(save_pending):
-		get_tree().process_frame.connect(save_pending, CONNECT_ONE_SHOT)
+	if auto_apply_changes and not get_tree().process_frame.is_connected(_save_pending):
+		get_tree().process_frame.connect(_save_pending, CONNECT_ONE_SHOT)
 
-func get_data(file_path: String, section: String, key :String, default :Variant = null) -> Variant:
-	var config := get_config(file_path)
-	return config.get_value(section, key, default)
+func save_high_score(key: String, value) -> void:
```

Comment:
I am not sure highscore functions belong here, this singleton is responsible for generic data storage, not minor details
the only specifics i allowed myself were a few FILE_PATH constants, and they are only there because i could not think of any other place currently

If we keep adding such minor functions to this singleton, it will eventually become polluted with tons of random crap from all over the codebase

Highscores could probably use a service of their own, there's probably more functionality to be moved into it from elsewhere

Reply:
I agree, high scores could be managed in another way. I would discuss this in different pull request after this one is merged.